Dockerfile

```diff
@@ -1,24 +1,48 @@
+# To build NeMo-Aligner from a base PyTorch container:
+#
+#   docker buildx build -t aligner:latest .
+#
+# To update NeMo-Aligner from a pre-built NeMo-Framework container:
+#
+#   docker buildx build --target=aligner-bump --build-arg=BASE_IMAGE=nvcr.io/nvidia/nemo:24.07 -t aligner:latest .
+#
 ARG BASE_IMAGE=nvcr.io/nvidia/pytorch:24.03-py3
 
-FROM ${BASE_IMAGE}
-ARG APEX_TAG=59b80ee8df79cec125794949327f29913c328746
-ARG TE_TAG=7d576ed25266a17a7b651f2c12e8498f67e0baea
-ARG MLM_TAG=a3fe0c75df82218901fa2c3a7c9e389aa5f53182  # On: core_r0.8.0
-ARG NEMO_TAG=e033481e26e6ae32764d3e2b3f16afed00dc7218  # On: r2.0.0rc1
+FROM ${BASE_IMAGE} AS aligner-bump
+WORKDIR /opt
+
 ARG ALIGNER_COMMIT=main
+# NeMo Aligner
+RUN <<"EOF" bash -exu
+if [[ ! -d NeMo-Aligner ]]; then
+    git clone https://github.com/NVIDIA/NeMo-Aligner.git
+    cd NeMo-Aligner
+    git checkout $ALIGNER_COMMIT
+    pip install --no-deps -e .
+    cd -
+fi
+cd NeMo-Aligner
+git fetch -a
+git checkout -f ${ALIGNER_COMMIT}
+git pull
```

Comment:
Any reason why not a `pip install --no-deps -e git+https://github.com/NVIDIA/NeMo-Aligner.git$ALIGNER_COMMIT` ?

Reply:
Great suggestion. I can try that

Reply:
Hm, so there are some weird nuances with using VCS editable installs.

1. looks like the directory will be the package name (nemo_aligner) and not the repository name (NeMo-Aligner)
2. repeated installs updates the local branch to the new remote ref, which appears to be equivalent to `git reset --hard origin/$new_ref`. This means when you `cd` into the cloned copy, it looks off:
```
root@dbc97c04ea58:/tmp/nemo-aligner# git status
On branch dev
Your branch and 'origin/dev' have diverged,
and have 241 and 28 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

nothing to commit, working tree clean
```

I think for now we should just land this and think about a better way to do this afterwards. For posterity, we need to do something like this:

```sh
 pip install --src=/opt --no-deps -e git+https://github.com/NVIDIA/NeMo-Aligner.git@${ALIGNER_COMMIT}#egg=nemo_aligner
```